test/programs/method-extreme/schema.json

```diff
@@ -0,0 +1,93 @@
+{
+  "type": "object",
+  "properties": {
+    "setTime": {
+      "parameters": [
+        {
+          "name": "d",
+          "type": "union",
+          "typeArguments": [
```

Comment:
@domoritz: [TypeScript -  generics](https://www.typescriptlang.org/docs/handbook/generics.html)
- the `string` in `Array<string>` is a `typeArgument`
- the `implements` in `T implements Array<string>` makes `Array<string>`  a `constraint` (on `T`)
- for `union`-types and `intersection`-types in method-definitions i used `typeArguments` to represent the types that are part of that union/intersection.

I unfortunately don't really have sources on how a function should be represented in a JSON-Schema, because there barely a sources. I've read somewhere that JSON-Schema isn't really meant to include functions, so i somewhat came up with my own way to represent them.

If there happens to be standards for this, that i just overlooked, i'm more than happy to follow it